janitor/ml.py

```diff
@@ -1,73 +1,73 @@
-""" Machine learning specific functions. """
-
-import unicodedata
-from typing import Any, Iterable, List, Tuple, Union
-
-import numpy as np
-import pandas as pd
-import pandas_flavor as pf
-
-from .utils import deprecated_alias
-
-
-@pf.register_dataframe_method
-@deprecated_alias(
-    target_columns="target_column_names",
-    feature_columns="feature_column_names",
-)
-def get_features_targets(
-    df: pd.DataFrame,
-    target_column_names: Union[str, Union[List, Tuple], Any],
-    feature_column_names: Union[str, Iterable[str], Any] = None,
-):
-    """
-    Get the features and targets as separate DataFrames/Series.
-
-    This method does not mutate the original DataFrame.
-
-    The behaviour is as such:
-
-    - ``target_column_names`` is mandatory.
-    - If ``feature_column_names`` is present, then we will respect the column
-        names inside there.
-    - If ``feature_column_names`` is not passed in, then we will assume that
-    the rest of the columns are feature columns, and return them.
-
-    Functional usage example:
-
-    .. code-block:: python
-
-        X, y = get_features_targets(df, target_column_names="measurement")
-
-    Method chaining example:
-
-    .. code-block:: python
-
-        import pandas as pd
-        import janitor.ml
-        df = pd.DataFrame(...)
-        target_cols = ['output1', 'output2']
-        X, y = df.get_features_targets(target_column_names=target_cols)
-
-    :param df: The pandas DataFrame object.
-    :param str/iterable target_column_names: Either a column name or an
-        iterable (list or tuple) of column names that are the target(s) to be
-        predicted.
-    :param str/iterable feature_column_names: (optional) The column name or
-        iterable of column names that are the features (a.k.a. predictors)
-        used to predict the targets.
-    :returns: (X, Y) the feature matrix (X) and the target matrix (Y). Both
-        are pandas DataFrames.
-    """
-    Y = df[target_column_names]
-
-    if feature_column_names:
-        X = df[feature_column_names]
-    else:
-        if isinstance(target_column_names, (list, tuple)):  # noqa: W503
-            xcols = [c for c in df.columns if c not in target_column_names]
-        else:
-            xcols = [c for c in df.columns if target_column_names != c]
-
-        X = df[xcols]
-    return X, Y
+""" Machine learning specific functions. """
+
+import unicodedata
+from typing import Hashable, Iterable, List, Tuple, Union
+
+import numpy as np
+import pandas as pd
+import pandas_flavor as pf
+
+from .utils import deprecated_alias
+
+
+@pf.register_dataframe_method
```

Comment:
Can you see what happened with `ml.py`? I see `Hashable` has been changed, but all these lines are showing changes. Did the line endings change?

Reply:
That has to be the case, but the typing import is the only thing that changed. It was put through the same version of Black as is in requirements.txt, so I can't imagine what would have caused it.

Reply:
Apparently Black formatting on a few tests is what cause Azure to fail. Not sure what's going on here.

Reply:
Ah you know what, I was the one that initially refactored the function to `ml.py`, and it looks like in the process the line endings were changed to Windows style, but no one noticed because it was a bulk change. The rest of the files seem to be Unix style, and you just returned it to normal. I'm not sure `black` does anything about this.